pkg/provisioning/service.go

```diff
@@ -43,25 +48,63 @@ func NewService(
 	plService PipelineService,
 	connService ConnectorService,
 	procService ProcessorService,
+	pipelinesDir string,
 ) *Service {
 	return &Service{
 		db:               db,
 		logger:           logger.WithComponent("provisioning.Service"),
 		pipelineService:  plService,
 		connectorService: connService,
 		processorService: procService,
+		pipelinesPath:    pipelinesDir,
 	}
 }
 
-func (s *Service) ProvisionConfigFile(ctx context.Context, path string) ([]string, error) {
+func (s *Service) Init(ctx context.Context) error {
+	s.logger.Debug(ctx).Str("pipelinesDir", s.pipelinesPath).Msg("initializing the provisioning service")
+
+	var files []string
+	err := filepath.WalkDir(s.pipelinesPath, func(path string, fileInfo fs.DirEntry, err error) error {
+		if strings.HasSuffix(path, ".yml") {
+			files = append(files, path)
+		}
+		return nil
+	})
+	if err != nil {
+		return cerrors.Errorf("could not iterate through the pipelines folder %q: %w", s.pipelinesPath, err)
+	}
+
+	if len(files) == 0 {
+		s.logger.Warn(ctx).Str("pipelinesDir", s.pipelinesPath).Msg("configuration folder has no YAML files, no pipelines created")
+	}
+
+	var multierr error
+	var pipelines []string
+	for _, file := range files {
+		provPipelines, err := s.provisionConfigFile(ctx, file)
+		if err != nil {
+			multierr = multierror.Append(multierr, err)
+		}
+		if len(provPipelines) != 0 {
+			pipelines = append(pipelines, provPipelines...)
+		}
+	}
+	s.logger.Info(ctx).Int("count", len(pipelines)).Str("pipelinesDir", s.pipelinesPath).Str("pipelines", strings.Join(pipelines, ", ")).Msg("pipelines successfully provisioned")
+
+	s.deleteOldPipelines(ctx, pipelines)
```

Comment:
I think it deserves a log msg on its own.. something like:

> 	s.logger.Info(ctx).
		Int("count", len(deletedPls)).
		Strs("pipelines", deletedPls).
		Msg("old pipelines that were deleted")
WDYT? and do you mean returning this info from the `init()` function?

Reply:
I mean that `deleteOldPipelines` would have to return the number of deleted pipelines so we can even log it.

It's a matter of taste, I personally would prefer to have 1 info line that contains all the info about what was the result of pipeline provisioning, for example:
```
pipeline configs provisioned component=provisioning.Service created=3 deleted=2 pipelines_path=/Users/lovro/dev/conduitio/conduit/pipelines
```

Reply:
> 

I like that single line

Reply:
should I also add the created and deleted pipeline names? or just the number

Reply:
That's a good idea, we could list the pipeline names (or maybe IDs) instead of providing the count.

```
pipeline configs provisioned component=provisioning.Service created=[pipeline1 pipeline2] deleted=[old-pipeline] pipelines_path=/Users/lovro/dev/conduitio/conduit/pipelines
```